src/list-ext.satyg

```diff
@@ -43,27 +37,46 @@ module List : sig
   val init : 'a list -> ('a list) option
   val reverse-append : 'a list -> 'a list -> 'a list
   val reverse-map : ('a -> 'b) -> 'a list -> 'b list
+  val all : ('a -> bool) -> 'a list -> bool
+  val any : ('a -> bool) -> 'a list -> bool
+  val zip : 'a list -> 'b list -> ('a * 'b) list
+  val unzip : ('a * 'b) list -> 'a list * 'b list
+  val find : ('a -> bool) -> 'a list -> 'a option
+  val partition : ('a -> bool) -> 'a list -> 'a list * 'a list
+  val iterate : int -> ('a -> 'a) -> 'a -> 'a list
+  val repeat : int -> 'a -> 'a list
+  val cycle- : int -> 'a list -> 'a list
+  val acons : 'a -> 'b -> ('a * 'b) list -> ('a * 'b) list
+  val max : ('a Ord.t) implicit -> 'a list -> 'a option
+  val min : ('a Ord.t) implicit -> 'a list -> 'a option
+  val fold : 'b -> ('a -> 'b -> 'b) -> 'a list -> 'b
+
+  % unstable API (may subject to change in the future)
+  val fold-left : ('a -> 'b -> 'a) -> 'a -> 'b list -> 'a
+  val fold-lefti : (int -> 'a -> 'b -> 'a) -> 'a -> 'b list -> 'a
+  val fold-right : ('a -> 'b -> 'b) -> 'b -> 'a list -> 'b
+  val fold-left-adjacent : ('a -> 'b -> 'b option -> 'b option -> 'a) -> 'a -> 'b list -> 'a
+  val map-adjacent : ('a -> 'a option -> 'a option -> 'b) -> 'a list -> 'b list
+  val mapi-adjacent : (int -> 'a -> 'a option -> 'a option -> 'b) -> 'a list -> 'b list
+  val bubblesort : ('a Ord.t) implicit -> 'a list -> 'a list
+  val insertsort : ('a Ord.t) implicit -> 'a list -> 'a list
+  val mergesort : ('a Ord.t) implicit -> 'a list -> 'a list
+
+  % obsoleted
```

Comment:
1. Let's show warning messages when obsoleted functions are used.
2. nit: `deprecated` is more common?

Reply:
Same comment to option-ext.satyg

Reply:
> Let's show warning messages when obsoleted functions are used.

👍 
I am going to make another pull-request for this task instead of working in this PR. Does this work for you?

> nit: deprecated is more common?

🆗

Reply:
Yeah sounds good to me